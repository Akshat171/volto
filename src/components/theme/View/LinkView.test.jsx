import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import LinkView from './LinkView';

const mockStore = configureStore();

describe('LinkView', () => {
  it('renders a link view component for logged users', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LinkView
            token="1234"
            content={{
              title: 'Hello World!',
              description: 'Hi',
              remoteUrl: '/news',
            }}
          />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('renders a link view component for anonymous users', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LinkView
            content={{
              title: 'Hello World!',
              description: 'Hi',
              remoteUrl: '/news',
            }}
          />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
