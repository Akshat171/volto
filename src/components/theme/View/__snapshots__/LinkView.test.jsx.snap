// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkView renders a link view component for anonymous users 1`] = `null`;

exports[`LinkView renders a link view component for logged users 1`] = `
<div
  className="ui container"
  id="page-document"
>
  <h1
    className="documentFirstHeading"
  >
    Hello World!
  </h1>
  <p
    className="documentDescription"
  >
    Hi
  </p>
  <p>
    The link address is:
     
    <a
      className={null}
      href="/news"
      onClick={[Function]}
      target={null}
      title={null}
    >
      /news
    </a>
  </p>
</div>
`;
